.github/workflows/e2e_docker.yml

```diff
@@ -0,0 +1,65 @@
+name: E2E (docker)
+
+on:
+  push:
+  pull_request:
+    branches: [main]
+
+concurrency:
+  group: '${{ github.workflow }} @ ${{ github.event.pull_request.head.label ||
+    github.head_ref || github.ref }}'
+  cancel-in-progress: true
+
+jobs:
+  e2e:
+    if: github.ref == 'refs/heads/main' || github.event_name == 'pull_request'
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # pin@v2
```

Comment:
why pin?

Reply:
this workflow is based on the original synpress repo. and these dependencies are unchanged since I didn't want to mess with anything

Reply:
I see that in their example https://github.com/drptbl/synpress-setup-example/blob/master/.github/workflows/e2e_multi-res_docker.yml#L27

but I can't imagine why they need to do that. if you want to pin at v2 it's simply,
```
uses: actions/checkout@v2
```

Please use version numbers instead of shas

Reply:
updated. although the shas didnt match with the tags mentioned in the comments. I updated them according to their actual value

Reply:
sorry, I should have been more specific: please use semver version specifiers. If it needs v2 then any `v2` will continue to work. The actions are well-tested. If the simpler version numbers don't work, we should find out sooner than later.

Reply:
noted. updated accordingly